### PR TITLE
Adds Flax BERT finetuning example on GLUE

### PR DESCRIPTION
**Pull request by marcvanzee**
# What does this PR do?

Adds Flax BERT finetuning example which finetunes on one of the GLUE tasks.

I evaluated all tasks 5 times and added the average runs, the best run. and stdev in a table in the README. I used the seed of the best run as the default.

I also ran all experiments on three devices: 8 Cloud TPU-v3, 1 Cloud TPU-v3, 1 P100 GPU. I compared the runtimes and put them in another table in the README.

This PR was discussed over Slack with @patrickvonplaten and @sgugger .

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [X] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: examples/flax/text-classification/README.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+<!---
+Copyright 2020 The HuggingFace Team. All rights reserved.
```
Comment (patrickvonplaten) [inline examples/flax/text-classification/README.md:2]:
```suggestion
Copyright 2021 The Google Flax Team Authors and HuggingFace Team. All rights reserved.
```

File: examples/flax/text-classification/run_flax_glue.py [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+      loss_fn: Function to compute the loss.
+    """
+
+    logits_fn: Callable = struct.field(pytree_node=False)
+    loss_fn: Callable = struct.field(pytree_node=False)
+
+
+@dataclass
+class FlaxTrainingArguments(TrainingArguments):
+    # We override these properties so we don't have to depend on Pytorch.
```
Comment (patrickvonplaten) [inline examples/flax/text-classification/run_flax_glue.py:133]: I guess I'm fine with it. @sgugger what do you think?

File: examples/flax/text-classification/run_flax_glue.py [hunk 3]
```diff
@@ -0,0 +1,476 @@
...
+    task = get_glue_task(data_args.task_name)
+
+    # Load pretrained model and tokenizer
+    config = BertConfig.from_pretrained(
+        model_args.config_name if model_args.config_name else model_args.model_name_or_path,
+        num_labels=len(task.labels),
+        finetuning_task=task.name,
+        cache_dir=model_args.cache_dir,
+    )
+    tokenizer = BertTokenizerFast.from_pretrained(
```
Comment (patrickvonplaten) [inline examples/flax/text-classification/run_flax_glue.py:386]:
```suggestion
    tokenizer = AutoTokenizer.from_pretrained(
```

File: examples/flax/text-classification/run_flax_glue.py [hunk 4]
```diff
@@ -0,0 +1,476 @@
...
+        model_args.tokenizer_name if model_args.tokenizer_name else model_args.model_name_or_path,
+        cache_dir=model_args.cache_dir,
+    )
+    model = FlaxBertForSequenceClassification.from_pretrained(
```
Comment (patrickvonplaten) [inline examples/flax/text-classification/run_flax_glue.py:390]:
```suggestion
    model = FlaxAutoModelForSequenceClassification.from_pretrained(
```
